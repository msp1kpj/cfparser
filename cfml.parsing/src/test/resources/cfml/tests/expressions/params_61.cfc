
component{
public void function test() {
  param string parse_error="but works";
  param type="string" name="parses_fine" default="works as expected";
  writeDump(parse_error);
  writeDump(parses_fine);
}
}