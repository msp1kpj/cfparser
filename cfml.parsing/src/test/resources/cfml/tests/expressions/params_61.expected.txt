/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
PUBLIC                   <public>
IDENTIFIER               <void>
FUNCTION                 <function>
IDENTIFIER               <test>
'('                      <(>
')'                      <)>
'{'                      <{>
PARAM                    <param>
STRING                   <string>
IDENTIFIER               <parse_error>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <but works>
CLOSE_STRING             <">
';'                      <;>
PARAM                    <param>
IDENTIFIER               <type>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <string>
CLOSE_STRING             <">
IDENTIFIER               <name>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <parses_fine>
CLOSE_STRING             <">
DEFAULT                  <default>
'='                      <=>
OPEN_STRING              <">
STRING_LITERAL           <works as expected>
CLOSE_STRING             <">
';'                      <;>
IDENTIFIER               <writeDump>
'('                      <(>
IDENTIFIER               <parse_error>
')'                      <)>
';'                      <;>
IDENTIFIER               <writeDump>
'('                      <(>
IDENTIFIER               <parses_fine>
')'                      <)>
';'                      <;>
'}'                      <}>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          (accessType public)
          (typeSpec (multipartIdentifier (identifier void)))
          function
          (identifier test)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (tagOperatorStatement
                (paramStatement
                  param
                  (paramExpression
                    (type string)
                    (multipartIdentifier (identifier parse_error))
                    =
                    (startExpression
                      (compareExpression
                        (baseExpression
                          (unaryExpression
                            (primaryExpression
                              (literalExpression (stringLiteral " (stringLiteralPart but works) "))
                            )
                          )
                        )
                      )
                    )
                  )
                  ;
                )
              )
            )
            (statement
              (tagOperatorStatement
                (paramStatement
                  param
                  (paramStatementAttributes
                    (param
                      (multipartIdentifier (identifier type))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression
                                (literalExpression (stringLiteral " (stringLiteralPart string) "))
                              )
                            )
                          )
                        )
                      )
                    )
                    (param
                      (multipartIdentifier (identifier name))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression
                                (literalExpression (stringLiteral " (stringLiteralPart parses_fine) "))
                              )
                            )
                          )
                        )
                      )
                    )
                    (param
                      (multipartIdentifier (identifier default))
                      =
                      (startExpression
                        (compareExpression
                          (baseExpression
                            (unaryExpression
                              (primaryExpression
                                (literalExpression (stringLiteral " (stringLiteralPart works as expected) "))
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                  ;
                )
              )
            )
            (statement
              (startExpression
                (compareExpression
                  (baseExpression
                    (unaryExpression
                      (memberExpression
                        (functionCall
                          (identifierOrReservedWord (identifier writeDump))
                          (
                          (argumentList
                            (argument
                              (startExpression
                                (compareExpression
                                  (baseExpression (unaryExpression (memberExpression (identifier parse_error))))
                                )
                              )
                            )
                          )
                          )
                        )
                      )
                    )
                  )
                )
              )
              ;
            )
            (statement
              (startExpression
                (compareExpression
                  (baseExpression
                    (unaryExpression
                      (memberExpression
                        (functionCall
                          (identifierOrReservedWord (identifier writeDump))
                          (
                          (argumentList
                            (argument
                              (startExpression
                                (compareExpression
                                  (baseExpression (unaryExpression (memberExpression (identifier parses_fine))))
                                )
                              )
                            )
                          )
                          )
                        )
                      )
                    )
                  )
                )
              )
              ;
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/